### PR TITLE
DOC: Updated documentation for numpy.squeeze

### LINKED ISSUE #16610: DOC: Update language on numpy.squeeze
In the [numpy.squeeze](https://numpy.org/doc/stable/reference/generated/numpy.squeeze.html) docstring, the use of "single-dimensional" and "singleton dimensions" (in the reference to numpy.expand_dims) is a bit confusing. This could be improved by using "one-dimensional" instead or rewording the docstring.

### PR DESCRIPTION
**Pull request by ruppysuppy**
Changed "single-dimensional" and "singleton dimensions" to "entries of length one" (fixes #16610)

### CODE REVIEW

**Comment by ruppysuppy** [comment]
I created a new branch but my previous commits are being displayed here. What can I do to avoid it?

**Comment by charris** [comment]
Did you create your branch off the master branch?

**Comment by ruppysuppy** [comment]
No, I made it from the last branch I was working on. Should I close this PR and create a new one after making a branch from the master?

**Comment by charris** [comment]
If you know how many commits are new, say `n`, you can do
```
$ git rebase HEAD~n --onto master
$ git push -f origin HEAD
```

**Comment by ruppysuppy** [comment]
Thanks a lot @charris :)

File: numpy/core/_add_newdocs.py
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py:3754]: I wonder if this would be clearer as either:
```suggestion
    Remove entries of value one from the shape of `a`.
```
or perhaps
```suggestion
    Remove axes of length one from `a`.
```

as written, this sort of implies `(n for n in a.shape if len(n) != 1)` when really it means `(n for n in a.shape if n != 1)`

**Comment by ruppysuppy** [comment]
@eric-wieser should I wait for the review by @melissawm or implement the changes suggested by you?

**Comment by eric-wieser** [comment]
Let's wait, there's no point you changing the wording twice. Thanks for the patch.

File: numpy/core/_add_newdocs.py
Comment (melissawm) [reply numpy/core/_add_newdocs.py:3754]: After consideration with the team we though the axes suggestion was the better approach, it's clearer and gets the point across. @ruppysuppy can you accept those changes? Thanks

**Review by melissawm — CHANGES_REQUESTED** [review]
Once the rewording is done I guess this is ready. Thanks!

**Review by ruppysuppy — COMMENTED** [review]
It seems like the build is failing for python 3.9-dev

**Review by melissawm — APPROVED** [review]
Not sure what's going on with the CI, but looks good to me :)

**Comment by rossbar** [comment]
> It seems like the build is failing for python 3.9-dev

This is a known (and unrelated) problem. Thanks @ruppysuppy and all the reviewers!

**Pull request merged**
